### DISCUSSION TITLE
Programmatically Adding and Retaining Highlights in ngx-extended-pdf-viewer

### DISCUSSION THREAD

**Discussion by aminasharaf** [stephanrauh/ngx-extended-pdf-viewer#2413] [Q&A]
In ngx-extended-pdf-viewer, we need to save all the highlights. When we log in next time, we should retain all previously highlighted data. Is it possible to add highlights programmatically? If yes, what are the values we need to maintain?

**Comment by stephanrauh** [comment]
You can save the entire document, either by asking the user to click the "download" button or by calling `await this.pdfViewerService.getCurrentDocumentAsBlob()`. The latter is documented here: https://pdfviewer.net/extended-pdf-viewer/export-file

Unfortunately, it's not possible to export and re-apply the highlight annotation using the low-level annotation API (https://pdfviewer.net/extended-pdf-viewer/export-annotations). To me, that's a nasty surprise, but in the short term, there's nothing I can do about that. You can export the highlight annotation, but it doesn't contain enough attributes to re-apply it.

Just out of curiosity: would you pay for this feature? I'm a consultant, so you could talk my boss into asking me to implement it.
